### PR TITLE
Make kube::util::ensure_clean_working_dir more verbose in log files

### PR DESCRIPTION
**Pull request by sttts**
Makes it easier to understand why files seem to be changed during CI build, especially
autogenerated files like conversions.

### CODE REVIEW

**Comment by cblecker** [comment]
Cross build is broken: https://github.com/kubernetes/kubernetes/issues/50675

File: hack/lib/util.sh [hunk 1]
```diff
@@ -485,9 +485,12 @@ kube::util::ensure_clean_working_dir() {
   while ! git diff HEAD --exit-code &>/dev/null; do
     echo -e "\nUnexpected dirty working directory:\n"
     git status -s | sed 's/^/  /'
```
Comment (cblecker) [inline hack/lib/util.sh]: This line happens no matter what, which outputs `git status -s` twice in interactive terminals

Comment (sttts) [inline hack/lib/util.sh]: oops, the line shouldn't be there.

Comment (sttts) [inline hack/lib/util.sh]: fixed

**Pull request merged**